src/Microsoft.OData.Core/ODataBatchWriter.cs

```diff
@@ -429,12 +372,9 @@ private void VerifyCanWriteStartBatch(bool synchronousCall)
         }
 
         /// <summary>
-        /// Starts a new batch - implementation of the actual functionality.
+        /// Starts a new batch.
         /// </summary>
-        private void WriteStartBatchImplementation()
-        {
-            this.SetState(BatchWriterState.BatchStarted);
-        }

```

Comment:
Observation is correct. Original design of this work item started from overriding all methods related to state but this one is the same and should be cleaned up.